plugins/admin_commands.py

```diff
@@ -2,70 +2,89 @@
+        parser.add_argument("-v", "--verbose", action="store_true")
+        parser.add_argument("-e", "--emulate", "--dryrun", action="store_true")
+
+        args = parser.parse_args(shlex.split(msg.content))
+
+        # Clamp the count. No negatives (duh), not more than 250 (don't get trigger happy)
+        args['count'] = min(max(args['count'], 0), 250)
+
+        if args['match']:
+            args['match'] = ' '.join(args['match'])
+
+        # find all possible members mentioned
         members = []
-        if len(args) > 3:
-            for s in args[3:]:
-                members.append(find_user(msg.guild, s))
+        if args['user']:
+            for q in args['user']:
+                u = find_user(msg.guild, q)
+                if u:
+                    members.append(u.id)
+
         await msg.delete()
-        # check if regexp version is used
-        args[0] = args[0].lower().endswith('rpurge') or args[0].lower().endswith('rprune')
-        deleted = await msg.channel.purge(limit=count,
-                                          check=((lambda x: self.rsearch(x, searchstr, members)) if
-                                                 args[0] else (lambda x: self.search(x, searchstr, members))))
-        fb = await respond(msg, f"**PURGE COMPLETE: {len(deleted)} messages purged.**")
+
+        if not args['emulate']:
+            # actual purging
+            deleted = await msg.channel.purge(limit=args['count'],
+                                              check=((lambda x: self.rsearch(x, args['match'], members)) if
+                                                     args['regex'] else
+                                                     (lambda x: self.search(x, args['match'], members))))
+        else:
+            # dry run to test your query
+            deleted = []
+            check = ((lambda x: self.rsearch(x, args['match'], members)) if args['regex'] else
+                     (lambda x: self.search(x, args['match'], members)))
+            async for m in msg.channel.history(limit=args['count']):
+                if check(m):
+                    deleted.append(m)
+
+        # if you REALLY want those messages
+        if args['verbose']:
+            await self.plugin_manager.hook_event("on_log_event", msg.guild,
+                                                 "**WARNING: Beginning verbose purge dump.**",
+                                                 log_type="purge_event")
+            for d in deleted[::-1]:
+                await self.plugin_manager.hook_event("on_log_event", msg.guild,
+                                                     d.content, log_type="purge_event")
+            await self.plugin_manager.hook_event("on_log_event", msg.guild,
+                                                 "**Verbose purge dump complete.**",
+                                                 log_type="purge_event")
+
+        fb = await respond(msg, f"**PURGE COMPLETE: {len(deleted)} messages purged.**" +
```

Comment:
Using a `+` and parentheses isn't necessary here

Reply:
```py
        # default Namespace built from parser defaults
        if namespace is None:
            namespace = RSNamespace()
```
Yes, because of this bit.

Reply:
The parentheses are very necessary because I use `"thing" if thing else ""`. If I remove the parentheses, it will just return "" instead of the whole string.

Reply:
Ah, yeah, I see that now.